### DISCUSSION TITLE
Add possibility to ignore a value in type-array-buffer-schema

### DISCUSSION THREAD

**Discussion by nasselk** [geckosio/geckos.io#275] [Ideas]
Hey, would it be possible to allow to ignore a value.

Here is an exemple : 

```js
const playerSchema = BufferSchema.schema("player", {
    id: uint16,
    x: float32,
    y: float32,
    angle: float32,
});


const positionSchema = BufferSchema.schema("positions", {
    players: [playerSchema],
});


const positionsModel = new Model(positionSchema);
```

```js
const positionsPack = {
   players: [],
};


//for each players
positionPack.push({
   id: player.id,
   x: player.x,
   //skip the y and angle because they haven't changed, so can save bandwith
});

sockets.raw.emit(positionsModel.toBuffer(positionsPack));
```

For now, if i do that and unpack it to get the real data, the values that are "undefined" get a very little value such as 1.0446119532155781e-40 for exemple. Could it be possible to keep them as undefined on output?

**Comment by reececomo** [comment]
Just in the spirit of problem solving:

### Alternative: 🗜️ **tinybuf**

Shameless self-plug, 🗜️ [**tinybuf**](https://www.npmjs.com/package/tinybuf) supports optional values, and is compatible with geckos.io/socket.io.

#### Equivalent code:

```ts
import { encoder, Type, Optional } from 'tinybuf';

const positionsModel = encoder({
  players: [
    {
      id: Type.UInt,
      x: Type.Float32,
      y: Optional(Type.Float32),
      angle: Optional(Type.Float32)
    }
  ]
});

sockets.raw.emit(PositionsModel.encode(positionsPack));
```

It also supports `undefined` and has additional float/scalar types for 1-byte and 2-byte floats (see [#Types](https://github.com/reececomo/tinybuf?tab=readme-ov-file#types)).

### Sanity check

You may might just want to check whether the optional/delta compression you're attempting is worth the effort/overhead. WebRTC data channels packets already have 60-120 bytes of overhead (due to SCTP + DTLS + UDP + IP), so a few bytes here and there may not really make a meaningful impact in the end.

Glenn Fiedler has some good advice around compression / delta encoding. See the various [gafferongames](https://gafferongames.com/post/serialization_strategies/) [techniques](https://gafferongames.com/post/snapshot_compression/) as examples.
